pkg/driver/config_dir_test.go

```diff
@@ -0,0 +1,304 @@
+		t.Errorf("symlink was not created at %s", etcAmazonEfs)
+	}
+
+	symlinkFile := path.Join(etcAmazonEfs, "foo.txt")
+	expectedLocation := path.Join(legacyDir, "foo.txt")
+
+	// create some file using the symlink
+	if f, err := os.Create(symlinkFile); err != nil {
+		t.Errorf("Unable to create file using symlink: %v", err)
+	} else {
+		f.Close()
+	}
+
+	// assert that the file was created in the legacy dir
+	if _, err := os.Stat(expectedLocation); err != nil {
+		t.Errorf("foo.txt was not created in the legacy directory '%s'", legacyDir)
+	}
+}
+
+// TestInitConfigDirPreferred asserts that a symlink is created to the preferred directory if efs-utils.conf is
+// not found in the legacy location.
+func TestInitConfigDirPreferred(t *testing.T) {
+	dir := tempDir(t)
+	defer os.RemoveAll(dir)
+
+	// create an empty legacy dir
+	legacyDir := filepath.Join(dir, "legacy")
+	if err := os.MkdirAll(legacyDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	// create the preferred dir
+	preferredDir := filepath.Join(dir, "preferred")
+	if err := os.MkdirAll(preferredDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	etcAmazonEfs := filepath.Join(dir, "symlink")
+
+	// function under test
+	if err := InitConfigDir(legacyDir, preferredDir, etcAmazonEfs); err != nil {
+		t.Fatalf("InitConfigDir returned an error: %v", err)
+	}
+
+	// assert symlink is there
+	if _, err := os.Lstat(etcAmazonEfs); err != nil {
+		t.Errorf("symlink was not created at %s", etcAmazonEfs)
+	}
+
+	symlinkFile := path.Join(etcAmazonEfs, "foo.txt")
+	expectedLocation := path.Join(preferredDir, "foo.txt")
+
+	// create some file using the symlink
+	if f, err := os.Create(symlinkFile); err != nil {
+		t.Errorf("Unable to create file using symlink: %v", err)
+	} else {
+		f.Close()
+	}
+
+	// assert that the file was created in the legacy dir
+	if _, err := os.Stat(expectedLocation); err != nil {
+		t.Errorf("foo.txt was not created in the preferred directory '%s'", legacyDir)
+	}
+}
+
+// TestInitConfigDirDoNothing asserts that a pre-existing symlink is not altered.
+func TestInitConfigDirDoNothing(t *testing.T) {
+	dir := tempDir(t)
+	defer os.RemoveAll(dir)
+
+	// create an empty legacy dir
+	legacyDir := filepath.Join(dir, "legacy")
+	if err := os.MkdirAll(legacyDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	// create the preferred dir
+	preferredDir := filepath.Join(dir, "preferred")
+	if err := os.MkdirAll(preferredDir, os.ModePerm); err != nil {
+		t.Fatalf("Unable to create directory: %v", err)
+	}
+
+	etcAmazonEfs := filepath.Join(dir, "symlink")
+
+	// create a symlink, as if this container has been run previously
+	if err := os.Symlink(preferredDir, etcAmazonEfs); err != nil {
+		t.Fatalf("unable to create symlink %v", err)
+	}
+
+	// function under test
+	if err := InitConfigDir(legacyDir, preferredDir, etcAmazonEfs); err != nil {
```

Comment:
What about a test case vEery similar to this where InitConfigDir is called twice? Arguably redundant, but in practice that is what will happen